File: bot.js
import * as THREE from "https://cdn.skypack.dev/three";

export class Bot {
  constructor(scene) {
    this.alive = true;
    this.health = 100;

    this.mesh = new THREE.Mesh(
      new THREE.BoxGeometry(1, 2, 1),
      new THREE.MeshStandardMaterial({ color: 0xff0000 })
    );
    this.mesh.position.set(
      Math.random() * 80 - 40,
      10,
      Math.random() * 80 - 40
    );
    scene.add(this.mesh);
  }

  update(player, zone) {
    if (!this.alive) return;

    // Move toward player
    this.mesh.position.lerp(player.mesh.position, 0.002);

    // Zone damage
    if (!zone.isInside(this.mesh.position)) {
      this.health -= 0.1;
    }

    if (this.health <= 0) {
      this.alive = false;
      this.mesh.visible = false;
    }
  }
}

File: bot.py
import random
import math

class Bot:
    def __init__(self):
        self.x = random.randint(100, 1180)
        self.y = random.randint(100, 620)
        self.health = 100
        self.alive = True
        self.radius = 8

    def update(self, player, zone):
        if not self.alive:
            return

        # Move toward player
        self.x += (player.x - self.x) * 0.002
        self.y += (player.y - self.y) * 0.002

        if not zone.is_inside(self.x, self.y):
            self.health -= 0.1

        if self.health <= 0:
            self.alive = False

    def draw(self, screen):
        if not self.alive:
            return
        import pygame
        pygame.draw.circle(screen, (255, 0, 0), (int(self.x), int(self.y)), self.radius)
